File: src/js/Libro.js
/**
 * @author javier garcia
 * Clase que representa un libro en la biblioteca.
 * @class
 */
class Libro {
  /**
   * Constructor de la clase Libro.
   * @constructor
   * @param {string} titulo - El título del libro.
   * @param {string} autor - El autor del libro.
   * @param {string} ISBN - El número ISBN del libro.
   * @param {string} estado - El estado del libro (disponible o reservado).
   */
  constructor(titulo, autor, ISBN, estado) {
    this.titulo = titulo;
    this.autor = autor;
    this.ISBN = ISBN;
    this.estado = estado;
  }

  /**
   * Obtiene el título del libro.
   * @returns {string} El título del libro.
   */
  getTitulo() {
    return this.titulo;
  }

  /**
   * Obtiene el autor del libro.
   * @returns {string} El autor del libro.
   */
  getAutor() {
    return this.autor;
  }

  /**
   * Obtiene el número ISBN del libro.
   * @returns {string} El número ISBN del libro.
   */
  getISBN() {
    return this.ISBN;
  }

  /**
   * Obtiene el estado del libro (disponible o reservado).
   * @returns {string} El estado del libro.
   */
  getEstado() {
    return this.estado;
  }

  /**
   * Establece el estado del libro (disponible o reservado).
   * @param {string} estado - El nuevo estado del libro.
   */
  setEstado(estado) {
    this.estado = estado;
  }

  /**
   * Convierte el objeto Libro en una representación de cadena de texto.
   * @returns {string} Una cadena de texto que representa el libro.
   */
  toString() {
    return `Libro { Título: ${this.titulo}, Autor: ${this.autor}, ISBN: ${this.ISBN}, Estado: ${this.estado} }`;
  }
}

// Exporta la clase para que pueda ser utilizada en otros módulos si es necesario.
export default Libro;

File: src/java/Libro.java
package java;

/**
 * @author javier garcia
 * Clase que representa un libro en la biblioteca.
 */
public class Libro {
    private String titulo;
    private String autor;
    private String ISBN;
    private EstadoLibro estado;

    /**
     * Constructor de la clase Libro.
     * @param titulo El título del libro.
     * @param autor El autor del libro.
     * @param ISBN El número ISBN del libro.
     * @param estado El estado del libro (disponible o reservado).
     */
    public Libro(String titulo, String autor, String ISBN, EstadoLibro estado) {
        this.titulo = titulo;
        this.autor = autor;
        this.ISBN = ISBN;
        this.estado = estado;
    }

    /**
     * Obtiene el título del libro.
     * @return El título del libro.
     */
    public String getTitulo() {
        return titulo;
    }

    /**
     * Obtiene el autor del libro.
     * @return El autor del libro.
     */
    public String getAutor() {
        return autor;
    }

    /**
     * Obtiene el número ISBN del libro.
     * @return El número ISBN del libro.
     */
    public String getISBN() {
        return ISBN;
    }

    /**
     * Obtiene el estado del libro (disponible o reservado).
     * @return El estado del libro.
     */
    public EstadoLibro getEstado() {
        return estado;
    }

    /**
     * Establece el estado del libro (disponible o reservado).
     * @param estado El nuevo estado del libro.
     */
    public void setEstado(EstadoLibro estado) {
        this.estado = estado;
    }

    /**
     * Convierte el objeto Libro en una representación de cadena de texto.
     * @return Una cadena de texto que representa el libro.
     */
    @Override
    public String toString() {
        return "Libro{" +
                "titulo='" + titulo + '\'' +
                ", autor='" + autor + '\'' +
                ", ISBN='" + ISBN + '\'' +
                ", estado=" + estado +
                '}';
    }
}